test/Sema/availability_versions.swift

```diff
@@ -1741,3 +1741,25 @@ func useHasUnavailableExtension(_ s: HasUnavailableExtension) {
   s.inheritsUnavailable() // expected-error {{'inheritsUnavailable()' is unavailable in macOS}}
   s.moreAvailableButStillUnavailable() // expected-error {{'moreAvailableButStillUnavailable()' is unavailable in macOS}}
 }
+
+@available(macOS 10.15, *)
+func f() -> Int { 17 }
+
+class StoredPropertiesWithAvailabilityInClosures {
+  private static let value: Int = {
```

Comment:
This could maybe use a test case asserting that we diagnose a reference to `f` when there is no `if #available`.

Reply:
Oh, we actually don't diagnose a reference to `f` there because it's an initializer, and those aren't exposed to the client (so they're checked at the deployment target)

Reply:
So then should f have availability later than the deployment target? Otherwise, we're not necessarily testing that TRCs get built out properly for the body.

Reply:
I have some test cases locally to check this, and will push them to a separate PR shortly.